github/private_registries.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2025 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// PrivateRegistriesService handles communication with the private registries
+// methods of the GitHub API.
+//
+// GitHub API docs: https://docs.github.com/rest/private-registries
+type PrivateRegistriesService service
+
+// PrivateRegistryType represents the type of private registry.
+type PrivateRegistryType string
+
+const (
+	PrivateRegistryTypeMavenRepository    PrivateRegistryType = "maven_repository"
+	PrivateRegistryTypeNugetFeed          PrivateRegistryType = "nuget_feed"
+	PrivateRegistryTypeGoProxyServer      PrivateRegistryType = "goproxy_server"
+	PrivateRegistryTypeNpmRegistry        PrivateRegistryType = "npm_registry"
+	PrivateRegistryTypeRubygemsServer     PrivateRegistryType = "rubygems_server"
+	PrivateRegistryTypeCargoRegistry      PrivateRegistryType = "cargo_registry"
+	PrivateRegistryTypeComposerRepository PrivateRegistryType = "composer_repository"
+	PrivateRegistryTypeDockerRegistry     PrivateRegistryType = "docker_registry"
+	PrivateRegistryTypeGitSource          PrivateRegistryType = "git_source"
+	PrivateRegistryTypeHelmRegistry       PrivateRegistryType = "helm_registry"
+	PrivateRegistryTypeHexOrganization    PrivateRegistryType = "hex_organization"
+	PrivateRegistryTypeHexRepository      PrivateRegistryType = "hex_repository"
+	PrivateRegistryTypePubRepository      PrivateRegistryType = "pub_repository"
+	PrivateRegistryTypePythonIndex        PrivateRegistryType = "python_index"
+	PrivateRegistryTypeTerraformRegistry  PrivateRegistryType = "terraform_registry"
+)
+
+// PrivateRegistryVisibility represents the visibility of a private registry.
+type PrivateRegistryVisibility string
+
+const (
+	PrivateRegistryVisibilityPrivate  PrivateRegistryVisibility = "private"
+	PrivateRegistryVisibilityAll      PrivateRegistryVisibility = "all"
+	PrivateRegistryVisibilitySelected PrivateRegistryVisibility = "selected"
+)
+
+// PrivateRegistry represents a private registry configuration.
+type PrivateRegistry struct {
```

Comment:
This struct is a configuration, so better name is:
```suggestion
type PrivateRegistryConfiguration struct {
```